### ISSUE TITLE
Arcane Mage - Hekili - Failed to make a recommendation

### ISSUE THREAD

**Issue by AleSSouza** [Hekili/hekili#2263]
### Before You Begin

- [X] I confirm that I have downloaded the latest version of the addon.
- [X] I am not playing on a private server.
- [X] I checked for an [existing, open ticket](https://github.com/Hekili/hekili/labels/recommendation) for this issue and was not able to find one.
- [X] I edited the title of this issue (above) so that it describes the issue I am reporting.
- [X] I am reporting an issue with the default priority included with the specialization (imported or edited priorities are not supported).

### WoW Version

Retail (Dragonflight)

### Describe the Issue

Doing the single target rotation on a dummy sometimes Hekili fail to recommend next action and make an automatic snapshot.

### How to Reproduce

1. Enter the game as Arcane Mage
2. Go to a practice dummy
3. Follow single target recommendations as it tells you 

### Talent Loadout

B4DAHVkVsxxZrVqBmAzlSR2f7TCIplIik0AJNJiIHISiESoFItkAAAAAAAAAAAAkkkkESSSyBA

### Snapshot

https://pastebin.com/HYFfsbXz

### Raidbots Sim Report (Link)

https://www.raidbots.com/simbot/report/ikoCCrFg4ndraeHvq8jnhm

**Comment by Hekili** [comment]
This is pretty bizarre.  Have you edited the priority or made some kind of changes?  There are lots of action list entries that are fully skipped in the snapshot that you provided.

**Comment by AleSSouza** [comment]
I have done nothing. Unfortunately.
Can u reproduce the error?
I have a trinket disabled (Irideus Fragment) because I macroed it.

**Comment by Hekili** [comment]
Can you show me a screenshot of your default action list?

/hekili > Priorities > Arcane > Action Lists > default, Entry #1 (or 2, 3, 4...)

**Comment by AleSSouza** [comment]
Like that?
![image](https://user-images.githubusercontent.com/87926384/211442375-b2183a4b-7ae9-45dc-a5ce-4779d63452c7.png)

**Comment by Hekili** [comment]
Don't expand the dropdown, just a screenshot of those settings.

**Comment by AleSSouza** [comment]
![image](https://user-images.githubusercontent.com/87926384/211446096-fe0acd36-8b07-491c-9f94-2d06219173de.png)
![image](https://user-images.githubusercontent.com/87926384/211446129-3fe9203f-80c8-47f2-8be1-1614fb78e0ad.png)
![image](https://user-images.githubusercontent.com/87926384/211446164-c7e0da3b-e54b-4e68-baea-fc1fae7537dd.png)
![image](https://user-images.githubusercontent.com/87926384/211446204-f095dbdc-02ce-434e-9d20-99e189d3af8d.png)
![image](https://user-images.githubusercontent.com/87926384/211446267-ac80d3df-c368-4a37-b6d7-155e7564c701.png)

Need more?

Ty

**Comment by Hekili** [comment]
Hm, still very strange.  The settings look okay here, but none of the abilities are being tested the Snapshot you provided.  Can you provide the Hekili.WTF file from your SavedVariables folder?

Does this occur only for Arcane?  Other Mage specs?  Other classes?

**Comment by AleSSouza** [comment]
Only arcane.
[Hekili.txt](https://github.com/Hekili/hekili/files/10378556/Hekili.txt)

I can reset settings to default. settings Just delete hekili.lua from wtf folder?

**Comment by Hekili** [comment]
You can reset on the Profiles tab.  I'm not sure what the problem is yet, though.

**Comment by AleSSouza** [comment]
Another snapshot with new profile.

https://pastebin.com/5KKTPwfV

**Comment by AleSSouza** [comment]
I'm reading through action lists and saw something weird:
- Time Warp is never recommended.
- Some entries of Meteor but we never use this talent/skill.

**Comment by Hekili** [comment]
The issue, in your case, is that the vast majority of the action list is never considered.

<details>
<summary>Example:</summary>

2.   use_items ( default - 10 )
Criteria for ??? FAIL at +0.04 - prev_gcd.1.arcane_surge[false]
 
3.   cancel_action:evocation ( default - 17 )
The action (cancel_action) is unusable at (0.04 + 0.00) because not channeling.
Time spent on this action:  0.06ms
TimeData:Arcane-default-17:cancel_action:0.06:Ability Known, Enabled(0.02):Post-TTR and Essential(0.03):Post Cycle(0.00):Post Usable(0.01)
 
4.   call_action_list:aoe_spark_phase ( default - 22 )
Criteria for aoe_spark_phase FAIL at +0.04 - talent.radiant_spark.enabled[true] & variable.aoe_spark_phase[false]

</details>

In the above, the first entry in the `default` action list that is tested is `default - 10`.  The next is `default - 17`.  The next is `default - 22`.  Everything of substance is being skipped when the addon is processing.  I'm not sure what settings one could use to trigger this issue.  I'm investigating, but this is really weird.

Normally, entries are skipped when they're explicitly disabled, but they don't appear to be disabled in your screenshot.  So I don't know yet how this is occurring.

**Comment by AleSSouza** [comment]
Something else I can do to help? Let me know.

**Comment by Hekili** [comment]
Can you confirm which profile name(s) (see /hekili > Profiles) you are using in the file you supplied?

**Comment by AleSSouza** [comment]
![image](https://user-images.githubusercontent.com/87926384/211674908-9dcca61a-ee71-412a-aac9-ad8e9e88e5d0.png)

**Comment by AleSSouza** [comment]
I tried a fresh WoW install and this keep happening.

**Comment by Hekili** [comment]
There is definitely no reason you'd need to do a fresh install.  I'm investigating and will update here when it is sorted.

**Comment by Noosferato** [comment]
I created mage char yesterday, have the same issue with Hekili addon. For other chars (hunter and warlock) Hekili works fine.

**Comment by wolfynsong** [comment]
I am also having issues with the Arcane rotation not suggesting certain abilities. I've never once had it suggest Time Warp, and Arcane Surge doesn't show up at all. I'm not yet 70 (64), which may be causing some issues, but I've had this skill for ages and it's never told me to use it, despite it being a pretty significant part of the burn phase from what I understand.

**Comment by Hekili** [comment]
@wolfynsong You almost certainly have not yet toggled your Cooldown abilities on.  See /hekili > Toggles.

**Comment by wolfynsong** [comment]
I see. Would have thought it belonged in the same cat as RoP/RS, but that does fix the issue. Thanks.

**Comment by Anastasia-Franco** [comment]
Well add me to the list. I also reset the profile and imported the latest simc I am only offered (during play time) Evocation, Rune of Power, Radiant Spark, then it spams Arcane Blast with an occasional Nether Tempest. And I will agree this addon is great for all of my other toons. It is literally killing me on Arcane Mage. Is there a special profile setup I need?  Yes cooldowns have been checked from the start.

**Comment by Hekili** [comment]
A snapshot is required to diagnose this issue.

## Snapshot

Please supply a Snapshot of the addon's decision-making when you are seeing this issue in-game. This is not a screenshot. To generate a Snapshot, please complete the following steps.

 * Log into your WoW character.
 * Recreate the issue you are reporting (i.e., generate 5 Holy Power).
 * When you see the recommendation you disagree with, press ALT-SHIFT-P to Pause and Snapshot (or use ALT-SHIFT-[ to Snapshot without pausing).
   * You can change these keybindings in /hekili > Toggles if needed.
   * If you take the snapshot at a different time, it will not be useful and you'll be asked to submit a new one.
 * Type /hekili and press Enter.
 * Open the Snapshots section on the left side.
 * Select the snapshot that shows the recommendation that you disagree with.
   * Snapshots are labeled with your specialization, the display in question (usually, you'll choose Primary), and the recommendations the addon made with their timings.
 * Click in the Export Snapshot box, press CTRL+A to select all text including hidden text and press CTRL+C to copy.
 * Open [https://pastebin.com](https://github.com/Hekili/hekili/issues/Pastebin), use CTRL+V to paste the snapshot text, then click Create New Paste.
 * Provide the Pastebin ***link*** in a new comment.

This step is essential, as most issues are related to specific classes, specializations, gear, talent choices, or other game systems.  If you do not provide this information, I cannot triage your problem.

**Comment by Anastasia-Franco** [comment]
The last priority update you made to CurseForge seemed to have done the
trick. Thank you

On 2023-04-06 19:32, Hekili wrote:

> A snapshot is required to diagnose this issue. 
> 
> SNAPSHOT
> 
> Please supply a Snapshot of the addon's decision-making when you are seeing this issue in-game. This is not a screenshot. To generate a Snapshot, please complete the following steps. 
> 
> * Log into your WoW character.
> * Recreate the issue you are reporting (i.e., generate 5 Holy Power).
> 
> * When you see the recommendation you disagree with, press ALT-SHIFT-P to Pause and Snapshot (or use ALT-SHIFT-[ to Snapshot without pausing). 
> 
> * You can change these keybindings in /hekili > Toggles if needed.
> * If you take the snapshot at a different time, it will not be useful and you'll be asked to submit a new one.
> 
> * Type /hekili and press Enter.
> * Open the Snapshots section on the left side.
> 
> * Select the snapshot that shows the recommendation that you disagree with. 
> 
> * Snapshots are labeled with your specialization, the display in question (usually, you'll choose Primary), and the recommendations the addon made with their timings.
> 
> * Click in the Export Snapshot box, press CTRL+A to select all text including hidden text and press CTRL+C to copy.
> * Open https://pastebin.com [1], use CTRL+V to paste the snapshot text, then click Create New Paste.
> * Provide the Pastebin _link_ in a new comment.
> 
> This step is essential, as most issues are related to specific classes, specializations, gear, talent choices, or other game systems. If you do not provide this information, I cannot triage your problem. 
> 
> --
> Reply to this email directly, view it on GitHub [2], or unsubscribe [3].
> You are receiving this because you commented.Message ID: ***@***.***>
 

Links:
------
[1] https://github.com/Hekili/hekili/issues/Pastebin
[2] https://github.com/Hekili/hekili/issues/2263#issuecomment-1499755968
[3]

**Comment by Hekili** [comment]
Thanks for the update.  It makes me suspect that the default priority version number was incorrect and not forcing an update.  I'll need to set up a better way to detect that going forward.

**Comment by gnarlyavelli** [comment]
Hello.

 I'm currently having the same issue that OP reported. 

Last night it was working fine but this afternoon I noticed it was recommending Touch of the Magi when it was on CD, and similarly would recommend Arcane Orb when it was on CD. Once the queue gets to that portion of the action list it bugs out and does not update. The icons don't even show the GCD timer of button presses, and at this point, I know what buttons to press without the addons help so I continue the rotation until it snaps back to normal. I tried a fresh install of the Addon and the issue still persists.
